DeviceSimulation/SimulationTests.cs

```diff
@@ -29,5 +40,190 @@ public void Should_Return_OK_Status()
+            Assert.True(dev1.Tags["IsSimulated"] == 'Y');
+        }
+
+        [Fact, Trait("Type", "IntegrationTest")]
+        public async void Should_Upgrade_Firmware_On_SimulatedDevice()
+        {
+            //delete the simulation that is already running
+            this.Should_Delete_Existing_Simulation();
+            var simulation = JObject.Parse(@"{   
+                'Enabled': true, 
+                'DeviceModels': [   
+                    {   
+                        'Id': 'truck-01', 
+                        'Count': 5 
+                    } 
+                ]
+            }");
+            var request = new HttpRequest(DS_ADDRESS + "/simulations");
+            request.AddHeader("Content-Type", "application/json");
+            request.SetContent(simulation);
+            var response = this.httpClient.PostAsync(request).Result;
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+
+            Thread.Sleep(15000);
+
+            ServiceClient serviceClient = ServiceClient.CreateFromConnectionString(IOTHUB_CONNECTION_STRING);
+            Task<CloudToDeviceMethodResult> directResponseFuture = serviceClient.InvokeDeviceMethodAsync("truck-01.0",
+                new CloudToDeviceMethod("FirmwareUpdate"));
+            CloudToDeviceMethodResult response1 = await directResponseFuture.ConfigureAwait(false);
+            Assert.Equal(200, response1.Status);
+        }
+
+        [Fact, Trait("Type", "IntegrationTest")]
+        public async void Should_Reboot_SimulatedDevice()
+        {
+            //delete the simulation that is already running
+            this.Should_Delete_Existing_Simulation();
+            var simulation = JObject.Parse(@"{   
+                'Enabled': true, 
+                'DeviceModels': [   
+                    {   
+                        'Id': 'chiller-01', 
+                        'Count': 5 
+                    } 
+                ]
+            }");
+            var request = new HttpRequest(DS_ADDRESS + "/simulations");
+            request.AddHeader("Content-Type", "application/json");
+            request.SetContent(simulation);
+            var response = this.httpClient.PostAsync(request).Result;
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+
+            Thread.Sleep(15000);
+
+            ServiceClient serviceClient = ServiceClient.CreateFromConnectionString(IOTHUB_CONNECTION_STRING);
+            Task<CloudToDeviceMethodResult> directResponseFuture = serviceClient.InvokeDeviceMethodAsync("chiller-01.0",
+                new CloudToDeviceMethod("Reboot"));
+            CloudToDeviceMethodResult response1 = await directResponseFuture.ConfigureAwait(false);
+
+            Assert.Equal(200, response1.Status);
+        }
+
+        [Fact, Trait("Type", "IntegrationTest")]
+        public void Should_Start_Given_Simulation()
+        {
+
+            var request1 = new HttpRequest(DS_ADDRESS + "/simulations/1");
```

Comment:
rename to more meaningful name. Same below for other variables with integers in name.